src/deferred.js

```diff
@@ -183,8 +186,13 @@ jQuery.extend( {
 								},
 
 								// Only normal processors (resolve) catch and reject exceptions
+								// Exceptions in other cases should not effect the promise result
 								process = special ?
-									mightThrow :
+									function() {
+										try {
+											mightThrow();
+										} catch ( e ) {}
```

Comment:
This was mainly just to test this out. I'm thinking that exceptions from a progress callback should be handled the same as exceptions in a success/error callback (~20 lines down). If we're ok with that then I think these two methods wrapping `mightThrow` can be combined and the `special` check will go within that single wrapper. Basically `special` just means that `mightThrow` throwing does not reject the promise...